### DISCUSSION TITLE
User Badges

### DISCUSSION THREAD

**Discussion by sgstair** [ludumdare/ludumdare#2031] [Ideas]
Design proposal for adding user badges:

Core idea: A badge will be an achievement for a user, potentially related to a specific event. General badges for users and badges related to the currently featured event will be displayed next to the user's name in places where the user is referenced - Typically displayed as a minimal icon with some mouseover text, possibly hover popup or a link to some kind of info page on the badge.

Proposed use:

The intent is to provide a visible reward for positive actions users take in the LD community, in an automated way. My initial proposal is a badge based on the "feedback karma" stat, and maybe something like bronze = 50 (127 games in LD39), silver = 100 (22 games in LD39), gold = 200 (1 game in LD39).

Design elements:
* Badge descriptions will be nodes. The content of the node will include directives for rendering the badges.
* Badges can have multiple promotion levels, e.g. level 0 = no badge, Level 1 = bronze, Level 2 = silver, Level 3 = gold (or similar)
* A new table containing: User id, Featured node id (nullable), Badge description id, badge promotion level, timestamp
* (Optionally) another table to track badge history
* Node fetch will add a "badges" key for "user" nodes, and fetch the global scope badges and featured event badges for users.
* Backend code to fetch / manipulate badges
* Badge assignment to be done in the magic (cron) script 
  * Minor caveat - We may need to split feedback karma into per-user rather than per-game for this calculation
* Client side code to fetch the relevant badge nodes from users and render badges when rendering usernames. (need to look into where exactly this would end up)
* New minimal API to query all badges for a user node
* New page or panel under user to show off all badges (as opposed to just the ones that are active), and maybe control visibility (via meta in user node)

**Comment by mikekasprzak** [comment]
Could be grouped in to this: #783

**Comment by sgstair** [comment]
yeah, it could possibly be merged with that. What I'm hoping for is commentary on the design ideas, and what should be done differently ;)

**Comment by mikekasprzak** [comment]
Ah, I didn't realize I didn't document my plans for trophies. I've updated #783 now with my original thoughts, and some new ones.

**Comment by mikekasprzak** [comment]
There is also #74, which may be loosely related. I originally had planned to make trophies an option you could add to the card, but it looks like I never actually said that. This might be an obsolete feature though.

**Comment by Iaknihs** [comment]
Following the Open Badges standard could let people take Badges along and show them off elsewhere, too.

Especially interesting for stuff like "Participation in X event" and even Top 3 Ranks or so.
